main.go

```diff
@@ -12,8 +14,28 @@ type ProcessSettings struct {
 	FlushAfter int
 }
 
+// ReceivedMessage is the message as received by a transport backend.
+type ReceivedMessage struct {
+	Payload          []byte
+	Attributes       map[string]string
+	ProviderMetadata interface{}
+}
+
 // FirehoseBackend is used for consuming messages from a transport and read/write to storage
 type FirehoseBackend interface {
+	Receive(ctx context.Context, numMessages uint32, visibilityTimeout time.Duration, messageCh chan<- ReceivedMessage) error
+
+	// NackMessage nacks a message on the queue
+	NackMessage(ctx context.Context, providerMetadata interface{}) error
+
+	// AckMessage acknowledges a message on the queue
+	AckMessage(ctx context.Context, providerMetadata interface{}) error
+
+	// RequeueFirehoseDLQ re-queues everything in the firehose queue
+	RequeueFirehoseDLQ(ctx context.Context, numMessages uint32, visibilityTimeout time.Duration) error
```

Comment:
I renamed this one as a new func that has to be defined here since the hedwig go equivalent requeues to the hedwig queue which we dont want right?

Reply:
Makes sense. Firehose will write to a dedicated DLQ